File: backend/src/Workora.Application/Features/AuditLogs/DTOs/AuditLogDto.cs
namespace Workora.Application.Features.AuditLogs.DTOs;

/// <summary>
/// DTO representing an audit trail log entry.
/// </summary>
public record AuditLogDto(
    int Id,
    int? UserId,
    string? ActorEmail,
    string Action,
    string EntityName,
    string? EntityId,
    string? OldValues,
    string? NewValues,
    string? IpAddress,
    string? UserAgent,
    DateTimeOffset Timestamp);

File: frontend/workora-ui/src/app/data/dtos/audit-log.dto.ts
export interface AuditLogDto {
  id: number;
  userId?: number | null;
  actorEmail?: string | null;
  action: string;
  entityName: string;
  entityId?: string | null;
  oldValues?: string | null;
  newValues?: string | null;
  ipAddress?: string | null;
  userAgent?: string | null;
  timestamp: string;
}
